app/assets/javascripts/admin/bulk_product_update.js.coffee

```diff
@@ -103,9 +106,16 @@ angular.module("ofn.admin").controller "AdminProductEditCtrl", ($scope, $timeout
     $scope.categoryFilter = "0"
     $scope.importDateFilter = "0"
 
+  $scope.$watch 'sortOptions', (sort) ->
+    if sort && sort.predicate != ""
+      $scope.sorting = sort.predicate + ' desc' if sort.reverse
+      $scope.sorting = sort.predicate + ' asc' if !sort.reverse
+      $scope.fetchProducts()
```

Comment:
this is copy pasted code so I'd extract it, maybe SortOptions itself would be a good place for it?

```ruby
  $scope.$watch 'sortOptions', (sort) ->
       return unless sort && sort.predicate != ""

       $scope.sorting = sort.getSortingExpr()
       $scope.fetchProducts()   
            
and then in sortOptions:
    getSortingExpr: () ->
         sortingExpr = this.predicate + ' desc' if this.reverse
         sortingExpr = this.predicate + ' asc' if !this.reverse
         sortingExpr
```

you could call this from the orders controller as well.